### PR TITLE
feat: add swc and tsup examples

### PR DESCRIPTION
**Pull request by MajesticString**
ESBuild isn't really designed to compile entire folders (it is meant to bundle files), so I made the SWC version as a replacement. It is just as fast.

### CODE REVIEW

**Comment by favna** [comment]
> ESBuild isn't really designed to compile entire folders

Right I should change that to `with-tsup`.. tsup uses esbuild and is much easier to configure and use globby in the background.

**Comment by MajesticString** [comment]
Should this PR be closed then?

**Comment by MajesticString** [comment]
Just added a tsup example with this PR

File: examples/with-tsup/tsup.config.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	format: ['cjs'],
+	minify: false,
+	tsconfig: 'tsconfig.json',
+	target: 'es2020',
+	splitting: false,
+	skipNodeModulesBundle: true,
+	sourcemap: true,
+	shims: false,
+	keepNames: true,
+	outDir: 'dist'
```
Comment (favna) [inline examples/with-tsup/tsup.config.ts:17]:
```suggestion
	clean: true,
	bundle: false,
	dts: false,
	entry: ['src/**/*.ts', '!src/**/*.d.ts'],
	format: ['cjs'],
	minify: false,
	tsconfig: 'tsconfig.json',
	target: 'es2020',
	splitting: false,
	skipNodeModulesBundle: true,
	sourcemap: true,
	shims: false,
	keepNames: true
```

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**